core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/BinderChildContextInitializer.java

```diff
@@ -0,0 +1,167 @@
+
+import org.springframework.aot.generate.GeneratedMethod;
+import org.springframework.aot.generate.GenerationContext;
+import org.springframework.aot.generate.MethodReference;
+import org.springframework.beans.factory.aot.BeanRegistrationAotContribution;
+import org.springframework.beans.factory.aot.BeanRegistrationAotProcessor;
+import org.springframework.beans.factory.aot.BeanRegistrationCode;
+import org.springframework.beans.factory.aot.BeanRegistrationExcludeFilter;
+import org.springframework.beans.factory.support.RegisteredBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.aot.ApplicationContextAotGenerator;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.core.log.LogAccessor;
+import org.springframework.javapoet.ClassName;
+import org.springframework.util.Assert;
+
+/**
+ * @author Chris Bono
+ * @since 4.0
+ */
+public class BinderChildContextInitializer implements ApplicationContextAware, BeanRegistrationAotProcessor, BeanRegistrationExcludeFilter {
+
+	private final LogAccessor logger = new LogAccessor(LogFactory.getLog(getClass()));
+	private DefaultBinderFactory binderFactory;
+	private final Map<String, ApplicationContextInitializer<ConfigurableApplicationContext>> childContextInitializers;
+	private volatile ConfigurableApplicationContext context;
+
+	public BinderChildContextInitializer() {
+		this.childContextInitializers = new HashMap<>();
+	}
+
+	public BinderChildContextInitializer(
+			Map<String, ApplicationContextInitializer<ConfigurableApplicationContext>> childContextInitializers) {
+		this.childContextInitializers = childContextInitializers;
+	}
+
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) {
+		Assert.isInstanceOf(ConfigurableApplicationContext.class, applicationContext);
+		this.context = (ConfigurableApplicationContext) applicationContext;
+	}
+
+	public void setBinderFactory(DefaultBinderFactory binderFactory) {
+		Assert.notNull(binderFactory, () -> "binderFactory must be non-null");
+		this.binderFactory = binderFactory;
+		if (!this.childContextInitializers.isEmpty()) {
+			this.logger.debug(() -> "Setting binder child context initializers on binder factory");
+			this.binderFactory.setBinderChildContextInitializers(this.childContextInitializers);
+		}
+	}
+
+	@Override
+	public boolean isExcluded(RegisteredBean registeredBean) {
+		return false;
+	}
+
+	@Override
+	public BeanRegistrationAotContribution processAheadOfTime(RegisteredBean registeredBean) {
+		if (registeredBean.getBeanClass().equals(getClass())) { //&& registeredBean.getBeanFactory().equals(this.context)) {
+			this.logger.debug(() -> "Beginning AOT processing for binder child contexts");
```

Comment:
It is better to not use lambda style, if you feed just plain string.
Saw similar things in other places, like `Assert.notNull(this.context, () -> "Unable to lookup binder factory from context as this.context is null");`

Reply:
Yeh, I went w/ consistency over performance but am happy to switch.